presto-main/src/main/java/io/prestosql/operator/CommonTableExpressionOperator.java

```diff
@@ -273,6 +295,46 @@ public boolean isFinished()
     @Override
     public void close() throws IOException
     {
+        if (snapshotState != null) {
+            snapshotState.close();
+        }
         LOG.debug("CTE(" + cteContext.getName() + ")[" + consumer + "-" + operatorInstaceId + "] Operator Closed");
     }
+
+    @Override
+    public Object capture(BlockEncodingSerdeProvider serdeProvider)
+    {
+
+        CommonTableOperatorState myState = new CommonTableOperatorState();
+        myState.operatorContext = operatorContext.capture(serdeProvider);
+        myState.cteContext = cteContext.capture(serdeProvider);
```

Comment:
only leader can restore the CTEContext; other operators cannot.

Reply:
Added a check for the same